pyartifactory/objects.py

```diff
@@ -231,6 +170,103 @@ def unlock(self, name: str) -> None:
         logger.debug("User % successfully unlocked", name)
 
 
+class ArtifactoryPermission(ArtifactoryObject):
+    """Models an artifactory permission."""
+
+    def __init__(self, artifactory: AuthModel) -> None:
+        super().__init__(artifactory)
+        if self._api_version == 2:
+            self._uri = "v2/security/permissions"
+        if self._api_version == 1:
+            self._uri = "security/permissions"
+
+    @overload
+    def create(self, permission: Permission,) -> Permission:
+        ...
+
+    @overload
+    def create(self, permission: PermissionV2,) -> PermissionV2:
+        ...
+
+    def create(self, permission: AnyPermission) -> AnyPermission:
+        """
+        Creates a permission
+        :param permission: Permission v2 or v1 object
+        :return: Permission v2 or v1
+        """
+        permission_name = permission.name
+        try:
+            self.get(permission_name)
+            logger.debug("Permission %s already exists", permission_name)
+            raise PermissionAlreadyExistsException(
+                f"Permission {permission_name} already exists"
+            )
+        except PermissionNotFoundException:
+            self._put(f"api/{self._uri}/{permission_name}", json=permission.dict())
```

Comment:
Now that we use aliases in our models we need to update the way we deserialize it:
```suggestion
            self._put(f"api/{self._uri}/{permission_name}", json=permission.dict(by_alias=True))
```